### PR TITLE
Duo parser fixes

### PR DESCRIPTION
**Pull request by giorgosp**
## Background

Fixes on Duo parsers

## Changes

- Fix validate:required tag
- Make `pantherlog.String` unmarshall json bool as string. Needed to parse some dubious booleans for Duo authentication logs

## Testing

- unit

### CODE REVIEW

File: internal/log_analysis/log_processor/parsers/duologs/administrator.go [hunk 1]
```diff
@@ -27,10 +27,10 @@ type AdministratorLog struct {
...
 	Timestamp    pantherlog.Time   `json:"timestamp" tcodec:"unix" description:"Unix timestamp of the event."`
-	UserName     pantherlog.String `json:"username" panther:"required" description:"The full name of the administrator who performed the action in the Duo Admin Panel. If the action was performed with the API this will be \"API\". Automatic actions like deletion of inactive users have \"System\" for the username. Changes synchronized from Directory Sync will have a username of the form (example) \"AD Sync: name of directory\"."`
+	UserName     pantherlog.String `json:"username" validate:"required" description:"The full name of the administrator who performed the action in the Duo Admin Panel. If the action was performed with the API this will be \"API\". Automatic actions like deletion of inactive users have \"System\" for the username. Changes synchronized from Directory Sync will have a username of the form (example) \"AD Sync: name of directory\"."`
```
Comment (kostaspap) [inline internal/log_analysis/log_processor/parsers/duologs/administrator.go:35]: You could include the new `panther:"username"` indicator :)

File: internal/log_analysis/log_processor/parsers/duologs/authentication.go [hunk 2]
```diff
@@ -31,12 +31,12 @@ type AuthenticationLog struct {
 	Email        pantherlog.String `json:"email" description:"The email address of the user, if known to Duo, otherwise none."`
```
Comment (kostaspap) [inline internal/log_analysis/log_processor/parsers/duologs/authentication.go:31]: You could add the new `panther:"email"` indicator

File: internal/log_analysis/log_processor/parsers/duologs/authentication.go [hunk 3]
```diff
@@ -31,12 +31,12 @@ type AuthenticationLog struct {
...
 	Result       pantherlog.String `json:"result" description:"The result of the authentication attempt. One of: \"SUCCESS\", \"FAILURE\", \"ERROR\", or \"FRAUD\"."`
 	Timestamp    pantherlog.Time   `json:"timestamp" tcodec:"unix" description:"Unix timestamp of the event."`
-	TxID         pantherlog.String `json:"txid" panther:"required" description:"The transaction ID of the event."`
+	TxID         pantherlog.String `json:"txid" validate:"required" description:"The transaction ID of the event."`
```
Comment (kostaspap) [inline internal/log_analysis/log_processor/parsers/duologs/authentication.go:39]: You could add this to the `panther:"trace_id"` indicator

File: internal/log_analysis/log_processor/parsers/duologs/authentication.go [hunk 4]
```diff
@@ -31,12 +31,12 @@ type AuthenticationLog struct {
...
 	User         User              `json:"user" description:"Information about the authenticating user."`
```
Comment (kostaspap) [inline internal/log_analysis/log_processor/parsers/duologs/authentication.go:40]: I am thinking this and `Alias` could be added to the `panther:"user_name"` indicator?

File: internal/log_analysis/log_processor/parsers/duologs/offline_enrollment.go [hunk 5]
```diff
@@ -26,10 +26,10 @@ const TypeOfflineEnrollment = "Duo.OfflineEnrollment"
...
 	Description  pantherlog.String `json:"description" description:"Information about the Duo Windows Logon client system as reported by the application."`
-	ISOTimestamp pantherlog.Time   `json:"isotimestamp" panther:"required" event_time:"true" tcodec:"rfc3339" description:"ISO8601 timestamp of the event."`
-	Object       pantherlog.String `json:"object" panther:"required" description:"The Duo Windows Logon integration's name."`
+	ISOTimestamp pantherlog.Time   `json:"isotimestamp" validate:"required" event_time:"true" tcodec:"rfc3339" description:"ISO8601 timestamp of the event."`
+	Object       pantherlog.String `json:"object" validate:"required" description:"The Duo Windows Logon integration's name."`
 	Timestamp    pantherlog.Time   `json:"timestamp" tcodec:"unix" description:"Unix timestamp of the event."`
-	UserName     pantherlog.String `json:"username" panther:"required" description:"The Duo username."`
+	UserName     pantherlog.String `json:"username" validate:"required" description:"The Duo username."`
```
Comment (kostaspap) [inline internal/log_analysis/log_processor/parsers/duologs/offline_enrollment.go:32]: `panther:"user_name"`

**Review by kostaspap — APPROVED** [review]
Nice!

File: internal/log_analysis/log_processor/parsers/duologs/administrator.go (hunk 1 above)
Comment (giorgosp) [reply internal/log_analysis/log_processor/parsers/duologs/administrator.go]: thanks, added all the indicators

**Pull request merged**